src/Oscoin/Crypto/Address/Internal.hs

```diff
@@ -0,0 +1,114 @@
+{-# LANGUAGE UndecidableInstances #-}
+module Oscoin.Crypto.Address.Internal where
+
+import           Oscoin.Prelude
+
+import           Oscoin.Configuration
+import           Oscoin.Crypto.Hash
+import           Oscoin.Crypto.PubKey
+
+import           Codec.CBOR.ByteArray (ByteArray)
+import           Data.Tagged
+
+{------------------------------------------------------------------------------
+  Types
+------------------------------------------------------------------------------}
+
+-- | An 'Address' is broken down as follows:
+--
+-- * /5/ bytes for the version prefix, so allocated:
+--
+-- @| 00 | 00 | 00 | 00 | 00 |
+--   ^    ^     ^   ^    ^
+--   |    |     |   |    |_ reserved (unused)
+--   |    |     |   |_ reserved (unused)
+--   |    |     |_ address format (only cbor for now)
+--   |    |_ address type (mainnet, testnet, devnet, multisig, etc)
+--   |_  protocol version (bumped on hard forks)@
+--
+-- * /23/ bytes for the CBOR-encoded address payload
+--
+-- * /6/ bytes for the checksum, computed from the @prefix + payload@.
+--
+data Address c = Address
+    { addressPrefix  :: AddressPrefix
+    , addressPayload :: AddressPayload c
+    } deriving (Eq, Show)
+
+data AddressPrefix = AddressPrefix
+    { protocolVersion :: ProtocolVersion
+    , addressType     :: AddressType
+    , addressFormat   :: AddressFormat
+    } deriving (Eq, Show)
+
+-- | A 'ProtocolVersion' is a byte representing the current version of the
+-- Oscoin protocol running. This makes possible to issue forks of the system
+-- by simply acting on this value, so that nodes running the old protocol
+-- won't be able to use addresses from the new fork. Conversely, nodes
+-- running the fork can be programmed to support both formats.
+newtype ProtocolVersion =
+    ProtocolVersion ProtocolVersionTag
+    deriving (Eq, Show)
+
+data ProtocolVersionTag =
+      ProtocolVersion_V1
```

Comment:
If we have an underscore here, we should make it consistent and have `AddressFormatTag_CBOR` no? Or remove the underscore here.

Reply:
Right, this is one of those hlint controversy: I tend to like the underscore better in these cases as at glance it would be easy to scan the different suffixes. However, when I did introduce the CBOR one, hlint complained so I did change it to camelcase.. and for some reason hlint didn't bark for these other 🤦‍♂️ .

Back to your original question: my pref would be underscores all the way down (with Hlint annotations to suppress the warnings) but I appreciate this might not be everyones' cup of tea, so please vote 😅

Reply:
Can we generally tell hlint to shutup about this? I think underscores for this makes sense, otherwise ItsALotOfCamelCase

Reply:
my hlint-scripting-foo is fairly weak, but I can look up something on ~StackOverflow~ the manual, to silence the entire "use camelcase" rule (should be fairly trivial , jokes aside).

Reply:
35cf933